partcad/src/partcad/sketch_factory_enrich.py

```
@@ -84,13 +101,50 @@ def __init__(self, ctx, source_project, target_project, config):
+                    raise Exception(
+                        "Attempting to parametrize a sketch that has no parameters: %s" % str(augmented_config)
+                    )
+                for param in sketch.config["with"]:
+                    if param not in augmented_config["parameters"]:
+                        raise Exception(
+                            "Attempting to parametrize a sketch with an unknown parameter: %s:%s: %s"
+                            % (self.source_project_name, self.source_sketch_name, param)
+                        )
+                    augmented_config["parameters"][param]["default"] = eval(
+                        type(augmented_config["parameters"][param]["default"]).__name__
+                    )(sketch.config["with"][param])
```

Ruff (0.8.2)

Use of possibly insecure function; consider using `ast.literal_eval`

(S307)